smart_open/tests/test_s3.py

```diff
@@ -98,14 +106,12 @@ def setUp(self):
 
     def tearDown(self):
         smart_open.s3.DEFAULT_MIN_PART_SIZE = self.old_min_part_size
-        s3 = boto3.resource('s3')
-        cleanup_bucket(s3, delete_bucket=True)
```

Comment:
I think you should still call `cleanup_bucket` at the end of each test, to ensure that the tests are fully isolated from each other.